<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SearchWidget</class>
 <widget class="QWidget" name="SearchWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>989</width>
    <height>231</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="styleSheet">
   <string notr="true">.QWidget, QScrollArea, .QFrame {
background-color: white;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="HeaderFrame" name="headerFrame" native="true">
     <property name="expandable" stdset="0">
      <bool>false</bool>
     </property>
     <property name="title" stdset="0">
      <string>Search</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>974</width>
        <height>226</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel {
font: bold;
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>-1</number>
       </property>
       <property name="leftMargin">
        <number>8</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>8</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
       <item>
        <widget class="QFrame" name="searchFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>10</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <property name="labelAlignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="horizontalSpacing">
             <number>10</number>
            </property>
            <property name="verticalSpacing">
             <number>10</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="graphLabel">
              <property name="text">
               <string>Graph:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="TreeViewComboBox" name="graphCombo">
              <property name="sizeAdjustPolicy">
               <enum>QComboBox::AdjustToContents</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="scopeLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Scope:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="scopeCombo">
              <property name="sizeAdjustPolicy">
               <enum>QComboBox::AdjustToContents</enum>
              </property>
              <item>
               <property name="text">
                <string>Nodes &amp; Edges</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Nodes only</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Edges only</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="resultsStorageLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Store results in:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="resultsStorageCombo">
              <property name="sizeAdjustPolicy">
               <enum>QComboBox::AdjustToContents</enum>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="searchTermsLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Search for:</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QComboBox" name="searchTermACombo">
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToContents</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="operatorCombo">
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToContents</enum>
                </property>
                <item>
                 <property name="text">
                  <string>=</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>≠</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>&gt;</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>≥</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>&lt;</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>≤</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>starts with</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ends with</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>contains</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="searchTermBCombo">
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToContents</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="customValueEdit">
                <property name="styleSheet">
                 <string notr="true">QLineEdit {
border: 1px solid #C9C9C9;
background-color: white;
color: black;
}</string>
                </property>
                <property name="placeholderText">
                 <string>Enter custom value...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Selection mode:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="selectionModeCombo">
              <property name="sizeAdjustPolicy">
               <enum>QComboBox::AdjustToContents</enum>
              </property>
              <item>
               <property name="text">
                <string>Replace current selection</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Add to current selection</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Remove from current selection</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="searchButton">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QPushButton {
font: bold;
}</string>
              </property>
              <property name="text">
               <string>Search</string>
              </property>
              <property name="icon">
               <iconset resource="../resources/GraphPerspective.qrc">
                <normaloff>:/tulip/graphperspective/icons/22/edit-find.png</normaloff>:/tulip/graphperspective/icons/22/edit-find.png</iconset>
              </property>
              <property name="shortcut">
               <string>Return</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="caseSensitivityCheck">
              <property name="text">
               <string>Case sensitive</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="resultsCountLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">font: normal;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HeaderFrame</class>
   <extends>QWidget</extends>
   <header>HeaderFrame.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TreeViewComboBox</class>
   <extends>QComboBox</extends>
   <header>tulip/TreeViewComboBox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/GraphPerspective.qrc"/>
  <include location="../../../../library/tulip-gui/resources/TulipGUIResource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>graphCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>graphIndexChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchTermACombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>termAChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchTermBCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>termBChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>customValueEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>SearchWidget</receiver>
   <slot>termBChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>699</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchButton</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>search()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>graphIndexChanged(int)</slot>
  <slot>termAChanged()</slot>
  <slot>termBChanged()</slot>
  <slot>search()</slot>
 </slots>
</ui>
